phoenix-core/src/main/java/org/apache/phoenix/compile/ProjectionCompiler.java

```diff
@@ -701,8 +723,7 @@ public Void visit(ProjectedColumnExpression expression) {
                              indexProjectedColumns.add(expression);
                              PColumn col = expression.getColumn();
                              // hack'ish... For covered columns with local indexes we defer to the server.
-                             if (col instanceof ProjectedColumn && ((ProjectedColumn) col)
-                                     .getSourceColumnRef() instanceof LocalIndexDataColumnRef) {
+                             if (context.getDataColumns().contains(col)) {
```

Comment:
Could you please explain this change? I am not sure about this one.

Reply:
Both the old and the new code checks wheter the column will be added from the base table by the coprocessor.

The old resolver returns a ProjectedColumn to a LocalIndexDataColumnRef, but the new resolver simply returns the projected column, so the logic wouldn't fire.

By checking if the column is in context.getDataColumns(), we check if a LocalIndexDataColumnRef has been created for this column, even if that is unwrapped by the time reach here.

Most of the non-test changes outside FromColumnCompiler are actually compensating for this change in the resolve behaviour. 

I wonder if there is a better way to change the resolver, so that we can get by with less changes to the code outside FromComplier.